codegen/style/generator.cpp

```diff
@@ -1186,14 +1186,15 @@ bool Generator::writeIconValues() {
 	for (auto i = iconMasks_.cbegin(), e = iconMasks_.cend(); i != e; ++i) {
 		QString filePath = i.key();
 		QByteArray maskData;
+		QFileInfo fileInfo(filePath);
 		if (filePath.startsWith("size://")) {
 			QStringList dimensions = filePath.mid(7).split(',');
 			if (dimensions.size() < 2 || dimensions.at(0).toInt() <= 0 || dimensions.at(1).toInt() <= 0) {
 				common::logError(common::kErrorFileNotOpened, filePath) << "bad dimensions";
 				return false;
 			}
 			maskData = iconMaskValueSize(dimensions.at(0).toInt(), dimensions.at(1).toInt());
-		} else if (QFileInfo(QStringView(filePath).split('-')[0].toString() + ".svg").exists()) {
+		} else if (QFileInfo(fileInfo.dir().absoluteFilePath(fileInfo.completeBaseName().split('-')[0] + ".svg")).exists()) {
```

Comment:
```suggestion
		} else if (QFileInfo(fileInfo.dir().absoluteFilePath(fileInfo.fileName().split('-')[0] + ".svg")).exists()) {
```
It's accepting the path without extension, completeBaseName strips by the last dot so will lead to bugs if the file name has a dot

Reply:
Addressed.